src/server/replica.cc

```diff
@@ -625,16 +650,23 @@ error_code Replica::SendNextPhaseRequest(string_view kind) {
   return std::error_code{};
 }
 
-error_code DflyShardReplica::StartFullSyncFlow(BlockingCounter sb, Context* cntx) {
+std::error_code DflyShardReplica::StartSyncFlow(BlockingCounter sb, Context* cntx,
+                                                std::vector<LSN>& lsns, bool& is_full_sync) {
   DCHECK(!master_context_.master_repl_id.empty() && !master_context_.dfly_session_id.empty());
 
   RETURN_ON_ERR(ConnectAndAuth(absl::GetFlag(FLAGS_master_connect_timeout_ms) * 1ms, &cntx_));
 
   VLOG(1) << "Sending on flow " << master_context_.master_repl_id << " "
           << master_context_.dfly_session_id << " " << flow_id_;
 
-  auto cmd = StrCat("DFLY FLOW ", master_context_.master_repl_id, " ",
-                    master_context_.dfly_session_id, " ", flow_id_);
+  std::string cmd;
+  if (lsns.size() > flow_id_ && master_context_.version > DflyVersion::VER1) {
+    cmd = StrCat("DFLY FLOW ", master_context_.master_repl_id, " ", master_context_.dfly_session_id,
+                 " ", flow_id_, " ", lsns[flow_id_]);
+  } else {
+    cmd = StrCat("DFLY FLOW ", master_context_.master_repl_id, " ", master_context_.dfly_session_id,
+                 " ", flow_id_);
```

Comment:
Since this is the exact prefix of the above, and we're certainly not on a hot path here, I'd start with `cmd` as this, and `StrAppend()` if needed. It's less error prone IMHO